### PR TITLE
feat(content-block-horizontal): introducing react wrapper

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#8622

### Description
This PR introduces the React Wrapper for Content Block Horizontal, both Storybook and Code Sandbox examples.

### Changelog

**New**

- created react wrapper for Content Block Horizontal

### CODE REVIEW

File: packages/web-components/src/components/content-block-horizontal/__stories__/README.stories.react.mdx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { Preview, Props, Description, Story } from '@storybook/addon-docs/blocks';
+import contributing from '../../../../../../docs/contributing-license.md';
+import { PropTypesRef } from '@carbon/ibmdotcom-web-components/es/components-react/content-block/content-block.js';
```
Comment (annawen1) [inline packages/web-components/src/components/content-block-horizontal/__stories__/README.stories.react.mdx:3]: should this be importing prop types from `content-block-horizontal`?

Comment (IgnacioBecerra) [reply packages/web-components/src/components/content-block-horizontal/__stories__/README.stories.react.mdx:3]: React wrappers seem to be unable to get the component proptypes if they are an extended from another one, but the Content Block Proptypes are able to be retrieved as normal

**Comment by annawen1** [comment]
@IgnacioBecerra looks like the react wrapper deploy preview is failing, can you take a look?

**Review by kennylam — APPROVED** [review]
LGTM!

**Pull request merged**